### PR TITLE
Create heroku apps with a valid name

### LINKED ISSUE #500: Heroku app naming tries to use underscore
Running suspenders example.com --heroku

will try to create heroku apps named example_com-production and example_com-staging but heroku only allows letters, numbers and hyphens not underscores so the app creation fails.

### PR DESCRIPTION
**Pull request by delphaber**
This should fix #500

### CODE REVIEW

**Comment by delphaber** [comment]
Ouch I have to please the hound

**Comment by tute** [comment]
This looks good to me. Please do honor Hound comments and rebase on top of latest master, so we merge.

File: lib/suspenders/app_builder.rb [hunk 1]
```diff
@@ -437,5 +437,9 @@ def port
     def serve_static_files_line
       "config.serve_static_files = ENV['RAILS_SERVE_STATIC_FILES'].present?\n"
     end
+
+    def heroku_app_name_for(environment)
+      "#{app_name.dasherize}-#{environment}"
```
Comment (jessieay) [inline lib/suspenders/app_builder.rb]: nice! I didn't know about `dasherize`

File: spec/support/suspenders.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module SuspendersTestHelpers
-  APP_NAME = 'dummy'
+  APP_NAME = 'dummy_app'
```
Comment (jessieay) [inline spec/support/suspenders.rb]: good idea on changing test app name!

**Comment by jessieay** [comment]
This is awesome! Do you have time to update, per Tute's comment @delphaber ? :rainbow:

**Comment by delphaber** [comment]
Hi, yep! My weekend just started ;)

**Comment by jessieay** [comment]
@delphaber Woo hoo, open source like its FRIDAY! :tada: :wine_glass: :dancer:

File: spec/features/heroku_spec.rb [hunk 3]
```diff
@@ -4,8 +4,9 @@
   scenario "Suspend a project for Heroku" do
     run_suspenders("--heroku=true")
 
-    expect(FakeHeroku).
-      to have_gem_included(project_path, "rails_stdout_logging")
+    %w(rails_stdout_logging).each do |gem|
```
Comment (tute) [inline spec/features/heroku_spec.rb]: Why the one element array here?

**Comment by tute** [comment]
Looking great, left a tiny question. Thanks for your work!

File: spec/features/heroku_spec.rb (hunk 3 above)
Comment (delphaber) [inline spec/features/heroku_spec.rb]: I was trying to keep the line < 80 chars and I thought that `have_gem_included` could be called more times in the future. Hound didn't like the `.\nto` too

Comment (tute) [inline spec/features/heroku_spec.rb]: Yeah I believe if it's too wide Hound would prefer:

```ruby
expect(FakeHeroku)
  .to have_gem_included(project_path, "rails_stdout_logging")
```

While it's only one gem I'd do it this way.

Comment (delphaber) [inline spec/features/heroku_spec.rb]: Oh I see. I'm gonna change that line! :)

**Comment by delphaber** [comment]
Done. No more hound notes ;)

File: spec/features/heroku_spec.rb (hunk 3 above)
Comment (tute) [inline spec/features/heroku_spec.rb]: Thanks for your work! But the one element array is still here.

**Comment by delphaber** [comment]
Sorry. My fault. It should be fine now :)

File: lib/suspenders/app_builder.rb (hunk 1 above)
Comment (CDainMiller) [inline lib/suspenders/app_builder.rb]: :+1:

**Comment by jessieay** [comment]
Merged at 7f2f94fa19bd2b3722a9bc166edc8597bde07d88

Thank you for your contribution!

Happy weekend! :hamster:

**Pull request closed without merging**